src/sentry/api/serializers/base.py

```diff
@@ -12,7 +12,9 @@ def serialize(objects, user=None, serializer=None):
 
     if not objects:
         return objects
-    elif not isinstance(objects, (list, tuple)):
+    # sets aren't predictable, so generally you should use a list, but it's
+    # supported out of convenience
+    elif not isinstance(objects, (list, tuple, set, frozenset)):
```

Comment:
I know we're not doing that for dicts either currently, but I had some good results in the past with paying the extra price to sort sets and dicts.  Makes them have a consistent order between machines and thus can be cached. Just a thing to note.